presto-spark-testing/src/test/java/com/facebook/presto/spark/testing/TestPrestoSparkLauncher.java

```diff
@@ -0,0 +1,370 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.testing;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.Session;
+import com.facebook.presto.hive.HiveHadoop2Plugin;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.QualifiedObjectName;
+import com.facebook.presto.spi.TableHandle;
+import com.facebook.presto.spi.connector.ConnectorFactory;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.hash.Hashing;
+import io.airlift.units.Duration;
+import org.joda.time.DateTimeZone;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Pattern;
+
+import static com.facebook.presto.spark.testing.Processes.destroyProcess;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tests.QueryAssertions.assertEqualsIgnoreOrder;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.google.common.io.Files.asByteSource;
+import static com.google.common.io.Files.write;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static com.google.common.io.Resources.getResource;
+import static com.google.common.io.Resources.toByteArray;
+import static java.lang.String.format;
+import static java.lang.Thread.sleep;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.Files.createDirectories;
+import static java.util.UUID.randomUUID;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.apache.hadoop.net.NetUtils.addStaticResolution;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * IMPORTANT!
+ * <p>
+ * Before running this test from an IDE, the project must be built with maven.
+ * <p>
+ * Please run:
+ * <p>
+ * ./mvnw clean install -pl presto-spark-launcher,presto-spark-package -am -DskipTests
+ * <p>
+ * from the project root after making any changes to the presto-spark-* codebase,
+ * otherwise this test may be running an old code version
+ */
+@Test(singleThreaded = true)
+public class TestPrestoSparkLauncher
```

Comment:
nit: `TestPrestoSparkLauncherIntegrationSmokeTest`? -- follow the name of `TestHiveIntegrationSmokeTest` :)